deepcell/graph_utils.py

```diff
@@ -0,0 +1,274 @@
+    """Extract feature vectors from cells in multiplexed imaging data
+
+    Args:
+        image (float): Multiplexed imaging dataset
+        label_image (int): Label image where every cell has been given a
+            unique integer id
+
+    Raises:
+        ValueError: if image and label_image batch size does not match
+
+    Returns:
+        np.array: matrix consisting of feature vectors for each cell
+    """
+    if image.shape[0] != label_image.shape[0]:
+        raise ValueError("image and label_image must have the same size for "
+                         "the batch dimmension")
+
+    # Find out the max number of cells
+    max_cells = get_max_cells(label_image)
+
+    # Instantiate the feature matrix
+    feature_matrix = np.zeros((image.shape[0], max_cells, image.shape[-1]))
+    label_matrix = np.zeros((label_image.shape[0], max_cells))
+
+    # Extract the features
+    # TODO: Add flag to include morphology features
+    for batch in range(image.shape[0]):
+        image_batch = image[batch]
+        label_image_batch = np.squeeze(label_image[batch])
+        for feature in range(image.shape[-1]):
+            props = regionprops_table(label_image_batch,
+                                      intensity_image=image_batch[..., feature],
+                                      properties=['mean_intensity', 'label'])
+
+            mean_intensity = props['mean_intensity']
+            labels = props['label']
+
+            feature_matrix[batch, 0:mean_intensity.shape[0], feature] = mean_intensity
+            label_matrix[batch, 0:labels.shape[0]] = labels
+
+    return feature_matrix
+
+
+def get_celltypes(label_image, celltype_image):
+    """Query cell type image with a given label image and get cell
+    type for each cell
+
+    Args:
+        label_image (np.array): Label image where every cell has been given a
+            unique integer id
+        celltype_image (np.array): Image where pixels belonging to each cell
+            are given an integer id specifying cell type
+
+    Raises:
+        ValueError: if image and label_image bach does not match
+
+    Returns:
+        np.array(float): Cell type matrix consisting of cell types for
+            each cell
+    """
+    if label_image.shape[0] != celltype_image.shape[0]:
+        raise ValueError("label_image and cell_type_image must have same size for "
+                         " the batch dimension")
+
+    # Find out max number of cells
+    max_cells = get_max_cells(label_image)
+
+    # Instantiate cell type matrix
+    celltype_matrix = np.zeros((label_image.shape[0], max_cells))
+
+    for batch in range(label_image.shape[0]):
+        label_image_batch = np.squeeze(label_image[batch])
+        celltype_image_batch = np.squeeze(celltype_image[batch])
+
+        props = regionprops_table(label_image_batch,
+                                  properties=['coords', 'label'])
+
+        for i in range(len(props['coords'])):
+            coords = props['coords'][i]
```

Comment:
You can iterate over both at once using `enumerate`

```suggestion
        for i, coords in enumerate(props['coords']):
```